src/main/scala/is/hail/methods/LinearRegression.scala

```diff
@@ -50,18 +50,18 @@ object LinearRegression {
     val (newVAS, inserter) = vds.insertVA(LinearRegression.schema, pathVA)
 
     vds.mapAnnotations { case (v, va, gs) =>
-
-      val (x: Vector[Double], isValid: Boolean) =
-        if (useDosages) {
-          val (x, mean) = RegressionUtils.dosageStats(gs, sampleMaskBc.value, n)
-          (x, n * mean >= combinedMinAC)
-        } else {
-          val (x, nHet, nHomVar, nMissing) = RegressionUtils.hardCallStats(gs, sampleMaskBc.value)
-          val ac = nHet + 2 * nHomVar
-          (x, !(ac < combinedMinAC || ac == 2 * (n - nMissing) || (ac == (n - nMissing) && x.forall(_ == 1))))
+      val (x: Vector[Double], ac) =
+        if (!useDosages) // replace by hardCalls in 0.2, with ac post-imputation
+          RegressionUtils.hardCallsWithAC(gs, n, sampleMaskBc.value)
```

Comment:
I say you're still trying too hard.  Just return a vector and take the sum.  Also, ac is a bit of a misnomer here.